### DISCUSSION TITLE
authentication-provider-work-github-gitpod-codespaces/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#616] [Comments]
# authentication-provider-work-github-gitpod-codespaces/

Azure / Microsoft 365 / Visual Studio Code / Speaker / Blogger

https://www.eliostruyf.com/authentication-provider-work-github-gitpod-codespaces/

**Comment by Leskd** [comment]
Hello, thank you for your share. your blog is amazing

follow you code, but I meet some issue when I want to use proxy,
1.  the proxy url don't open the vscode,  my url like this:   
```js
context.subscriptions.push(
    vscode.commands.registerCommand(`helloworld.signin`, async () => {
      // Get an externally addressable callback URI for the handler that the authentication provider can use
      const callbackUri = await vscode.env.asExternalUri(
        vscode.Uri.parse(`${vscode.env.uriScheme}://${extensionId}/auth-complete`)
      );

      // Add your code to integrate with an authentication provider here - we'll fake it.
      vscode.env.clipboard.writeText(callbackUri.toString());
      await vscode.env.openExternal(`http://localhost/redirect?state=${encodeURIComponent(callbackUri.toString(true))}` as any);
      await vscode.window.showInformationMessage(
        'Open the URI copied to the clipboard in a browser window to authorize.'
      );
    })
  );
}

export class UriHandler extends vscode.EventEmitter<vscode.Uri> implements vscode.UriHandler {
  async handleUri(uri: vscode.Uri): Promise<void> {
    console.log('🚀 ~ UriHandler ~ handleUri ~ uri:', uri);
    this.fire(uri);
  }
}
```
after I selected the signin command, open a window in Chrome. and url redirected.
but the extension don't recive any message.   
should I use `https`?

**Reply by Leskd** [reply]
Maybe code-server can have a special route it self to handle this message? Like below:

``https://code--proxy-server-url/protocol-handler?state=vscode://my-plugin/path``

This added the benefit: there're no notification/confirmation when user open it, more smooth than VSCode.

I want to do this.

**Reply by estruyf** [reply]
Hi @Leskd, what exactly does your local server do? Because the URI scheme to use is `vscode://` for opening your local VSCode instance.

If you check the [vscode-redirect](https://github.com/estruyf/vscode-redirect/blob/main/pages/index.tsx#L18-L25) sample, there is a check to verify if the URL contains the VSCode schema. If that is the case, it should automatically redirect you (now your browser will not automatically open VSCode; it will ask you to do it, as that is a security policy).

**Comment by ejfasting** [comment]
Hi, 

Great article! :)

I've been using the uriHandler, as you do in the [previous article](https://www.eliostruyf.com/devhack-vscode-uri-handler-codespaces/), but unfortunately there looks to be no way to write a simple 'you can now close this page' to the browser before/after the command is executed. 
This proxy-setup could potentially fix this, and tell the user to close the page (even though the following code exchange might not succeed), but it requires you to spin up something locally to receive the callback.

Do you by any chance know a way around this (spinning up the local server)? I'm considering trying to make this all work inside of a webview, so the user doesnt have to leave vscode at all.

//e

Its unfortunate that there is no 'easy to use' method of doing it,
